src/NuGet.Core/NuGet.Packaging/Signing/Signatures/RepositoryCounterSignature.cs

```
@@ -0,0 +1,77 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+#if IS_DESKTOP
+using System.Security.Cryptography.Pkcs;
+#endif
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    public class RepositoryCounterSignature : Signature, IRepositorySignature
+    {
+#if IS_DESKTOP
+        private readonly Uri _nuGetV3ServiceIndexUrl;
+        public Uri NuGetV3ServiceIndexUrl => _nuGetV3ServiceIndexUrl;
+
+        private readonly IReadOnlyList<string> _nuGetPackageOwners;
+        public IReadOnlyList<string> NuGetPackageOwners => _nuGetPackageOwners;
+
+
+        private RepositoryCounterSignature(SignerInfo counterSignerInfo, Uri nuGetV3ServiceIndexUrl)
+            : base(counterSignerInfo, SignatureType.Repository)
+        {
+            _nuGetV3ServiceIndexUrl = nuGetV3ServiceIndexUrl;
+            _nuGetPackageOwners = AttributeUtility.GetNuGetPackageOwners(SignerInfo.SignedAttributes);
+        }
+
+        public static RepositoryCounterSignature GetRepositoryCounterSignature(PrimarySignature primarySignature)
+        {
+            var counterSignatures = primarySignature.SignerInfo.CounterSignerInfos;
+            var repositoryCounterSignatures = new List<SignerInfo>();
+            Uri potentialNuGetV3ServiceIndexUrl = null;
+            Uri lastFoundNuGetV3ServiceIndexUrl = null;
+
+            // We only care about the repository countersignatures, not any kind of counter signature
+            foreach (var counterSignature in counterSignatures)
+            {
+                potentialNuGetV3ServiceIndexUrl = null;
+                try
+                {
+                    potentialNuGetV3ServiceIndexUrl = AttributeUtility.GetNuGetV3ServiceIndexUrl(counterSignature.SignedAttributes);
+                }
+                // This counter signature is not a valid repository countersignature
+                catch (SignatureException) { }
```

After you determine it's a repository countersignature, do not catch this exception.  Let the exception bubble up to indicate it's an invalid repository countersignature.